test/integration/controllermanager/shoot/state/state_suite_test.go

```diff
@@ -0,0 +1,130 @@
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	controllermanagerconfigv1alpha1 "github.com/gardener/gardener/pkg/controllermanager/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/shoot/state/finalizer"
+	"github.com/gardener/gardener/pkg/logger"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	gardenerenvtest "github.com/gardener/gardener/test/envtest"
+)
+
+func TestState(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "State Suite")
+}
+
+const (
+	testID                     = "shootstate-controller-test"
+	gardenClusterApiServerArgs = `--disable-admission-plugins=DeletionConfirmation,ResourceReferenceManager,ExtensionValidator,ShootQuotaValidator,ShootValidator,ShootTolerationRestriction,ShootDNS`
+)
+
+var (
+	ctx = context.Background()
+	log logr.Logger
+
+	testEnv    *gardenerenvtest.GardenerTestEnvironment
+	testClient client.Client
+	restConfig *rest.Config
+
+	mgr        manager.Manager
+	mgrContext context.Context
+	mgrCancel  context.CancelFunc
+
+	testNamespace *corev1.Namespace
+	testRunID     string
+)
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(logger.MustNewZapLogger(logger.DebugLevel, logger.FormatJSON, zap.WriteTo(GinkgoWriter)))
+	log = logf.Log.WithName(testID)
+
+	By("Start test environment")
+	testEnv = &gardenerenvtest.GardenerTestEnvironment{
+		GardenerAPIServer: &gardenerenvtest.GardenerAPIServer{
+			Args: []string{gardenClusterApiServerArgs},
+		},
+	}
+
+	var err error
+	restConfig, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(restConfig).NotTo(BeNil())
+
+	By("Create test client")
+	testClient, err = client.New(restConfig, client.Options{Scheme: kubernetes.GardenScheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Create test Namespace")
+	// create dedicated namespace for each test run,
+	// so that we can run multiple tests concurrently for stress tests
+	testNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "garden-",
+		},
+	}
+	Expect(testClient.Create(ctx, testNamespace)).To(Succeed())
+	log.Info("Created Namespace for test", "namespaceName", testNamespace.Name)
+	testRunID = testNamespace.Name
+
+	By("Setup manager")
+	mgr, err = manager.New(restConfig, manager.Options{
+		Scheme:  kubernetes.GardenScheme,
+		Metrics: metricsserver.Options{BindAddress: "0"},
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{testNamespace.Name: {}},
+		},
+		Controller: controllerconfig.Controller{
+			SkipNameValidation: ptr.To(true),
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Setup finalizer reconciler")
+	Expect((&finalizer.Reconciler{
+		Config: controllermanagerconfigv1alpha1.ShootStateControllerConfiguration{
+			ConcurrentSyncs: ptr.To(5),
+		},
+	}).AddToManager(mgr)).To(Succeed())
+
+	mgrContext, mgrCancel = context.WithCancel(ctx)
+	go func() {
+		defer GinkgoRecover()
+		fmt.Println("STARTING THE CONTROLLER MANAGER")
```

Comment:
This seems like a leftover.
```suggestion
```

Reply:
Thanks for spotting it, indeed it is. Removed it in my latest updates.